api/envoy/api/v2/route/route.proto

```diff
@@ -303,6 +303,9 @@ message CorsPolicy {
 
   // Specifies if CORS is enabled. Defaults to true. Only effective on route.
   google.protobuf.BoolValue enabled = 7;
+
+  // Specifies regex patterns that match allowed origins.
+  repeated string allow_origin_regex = 8;
```

Comment:
I think it's worth moving this up, next to `allow_origin` since they're releated (and add `// [#comment:next free field: 9]` just above the `message CorsPolicy`).

I think it's also woth a comment to indicate that an origin that matches either `allow_origin` or `allow_origin_regex` will be allowed.